iis-wf/iis-wf-import/src/main/java/eu/dnetlib/iis/wf/importer/infospace/truncator/DocumentMetadataAvroTruncator.java

```diff
@@ -0,0 +1,200 @@
+                    .collect(Collectors.toList());
+        }
+    }
+
+    public static class KeywordsTruncator implements Serializable {
+        private Function<List<CharSequence>, List<CharSequence>> listTruncator;
+        private Function<CharSequence, CharSequence> stringTruncator;
+
+        public KeywordsTruncator(int maxKeywordsSize, int maxKeywordLength) {
+            this.listTruncator = (Function<List<CharSequence>, List<CharSequence>> & Serializable) xs ->
+                    xs.stream().limit(maxKeywordsSize).collect(Collectors.toList());
+            this.stringTruncator = (Function<CharSequence, CharSequence> & Serializable) x ->
+                    StringTruncator.truncateWithoutWordSplit(x.toString(), maxKeywordLength);
+        }
+
+        public void setListTruncator(Function<List<CharSequence>, List<CharSequence>> listTruncator) {
+            this.listTruncator = listTruncator;
+        }
+
+        public void setStringTruncator(Function<CharSequence, CharSequence> stringTruncator) {
+            this.stringTruncator = stringTruncator;
+        }
+
+        List<CharSequence> truncate(List<CharSequence> xs) {
+            return listTruncator.apply(xs)
+                    .stream()
+                    .map(x -> stringTruncator.apply(x))
+                    .collect(Collectors.toList());
+        }
+    }
+
+    public static Builder newBuilder() {
+        return new Builder();
+    }
+
+    public static class Builder {
+        private Integer maxAbstractLength;
+        private Integer maxTitleLength;
+        private Integer maxAuthorsSize;
+        private Integer maxAuthorFullnameLength;
+        private Integer maxKeywordsSize;
+        private Integer maxKeywordLength;
+
+        public Builder setMaxAbstractLength(int maxAbstractLength) {
+            this.maxAbstractLength = maxAbstractLength;
+            return this;
+        }
+
+        public Builder setMaxTitleLength(Integer maxTitleLength) {
+            this.maxTitleLength = maxTitleLength;
+            return this;
+        }
+
+        public Builder setMaxAuthorsSize(Integer maxAuthorsSize) {
+            this.maxAuthorsSize = maxAuthorsSize;
+            return this;
+        }
+
+        public Builder setMaxAuthorFullnameLength(Integer maxAuthorFullnameLength) {
+            this.maxAuthorFullnameLength = maxAuthorFullnameLength;
+            return this;
+        }
+
+        public Builder setMaxKeywordsSize(Integer maxKeywordsSize) {
+            this.maxKeywordsSize = maxKeywordsSize;
+            return this;
+        }
+
+        public Builder setMaxKeywordLength(Integer maxKeywordLength) {
+            this.maxKeywordLength = maxKeywordLength;
+            return this;
+        }
+
+        public DocumentMetadataAvroTruncator build() {
+            DocumentMetadataAvroTruncator instance = new DocumentMetadataAvroTruncator();
+
+            if (Objects.nonNull(maxAbstractLength)) {
```

Comment:
If the object is currently instantiated by relying on the statically defined values within `ImportInformationSpaceConstants` class do we need to check the nullability of those fields here and in similar places below?

Reply:
With this implementation the check is necessary. `maxAbstractLength` in this null-check is an `Integer' field within builder nested class. If this field was set using `setMaxAbstractLength()` method then we use it when creating abstract truncator, if this field was not set and was null then we fallback to default abstract truncator with default value created by `DocumentMetadataAvroTruncator` class.

However, builder logic is now changed as a result of changing the way truncation parameters are passed to importer job and this null-check is not needed.  `DocumentMetadataAvroTruncator` is created with explicit parameters provided in the workflow.